File: AndroidXPatternsApp/jmvvm-app/src/main/java/com/woowrale/jmvvm/di/modules/ServicesModule.java
package com.woowrale.jmvvm.di.modules;

import com.woowrale.jmvvm.data.repository.ApiClient;
import com.woowrale.jmvvm.data.repository.ApiService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ServicesModule {

    @Provides
    @Singleton
    ApiService getApiService(){
        return new ApiClient().getClient().create(ApiService.class);
    }
}

File: AndroidXPatternsApp/kmvvm-app/src/main/java/com/woowrale/kmvvm/di/modules/ServicesModule.kt
package com.woowrale.kmvvm.di.modules

import com.woowrale.kmvvm.data.repository.ApiClient
import com.woowrale.kmvvm.data.repository.ApiService
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ServicesModule {

    @Provides
    @Singleton
    fun getApiServices():ApiService{
        return ApiClient.getclient().create(ApiService::class.java)
    }
}
